libmozevent/utils.py

```diff
@@ -175,16 +184,31 @@ class AsyncRedis(object):
     @staticmethod
     async def connect():
         if AsyncRedis.redis.get(None) is None:
+            retry_on_error = [
+                ConnectionError,
+                ExecAbortError,
+                PubSubError,
+                RedisError,
+                ResponseError,
+                TimeoutError,
+                WatchError,
+            ]
+
+            retries = 5
             AsyncRedis.redis.set(
                 await aioredis.from_url(
                     os.environ["REDIS_TLS_URL"],
                     decode_responses=False,
                     ssl_cert_reqs=None,
+                    retry_on_error=retry_on_error,
+                    retry=Retry(ExponentialBackoff(), retries),
                 )
                 if os.getenv("REDIS_TLS_URL", None) is not None
                 else await aioredis.from_url(
                     os.environ["REDIS_URL"],
                     decode_responses=False,
+                    retry_on_error=retry_on_error,
+                    retry=Retry(ExponentialBackoff(), retries),
```

Comment:
Let's use a longer cap and base for the exponential backoff, e.g. `ExponentialBackoff(cap=7, base=1)`